### PR TITLE
OSD-25868: Adds `aws-hosted-cp` as a valid Platform name

### PR DESCRIPTION
**Pull request by joshbranham**
## What does this PR do? / Related Issues / Jira

This extends the Platform structs to accept `aws-hosted-cp` as a valid value for looking up AWSHCP `ByName()`. This value is used internally still and blocks integrating the updated verifier.

## Checklist

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have tested the functionality against gcp / aws, it doesn't cause any regression
- [ ] I have added execution results to the PR's readme

## Reviewer's Checklist

- [ ] (This needs to be done after technical review) I've run the branch on my local, verified that the functionality is ok

### CODE REVIEW

File: pkg/data/cloud/platform_test.go [hunk 1]
```diff
@@ -123,6 +123,10 @@ func TestByName(t *testing.T) {
 			name: "aws-hcp",
 			want: AWSHCP,
 		},
```
Comment (abyrne55) [inline pkg/data/cloud/platform_test.go:125]: Can we also add a test case for an empty string for `name`? Should return an empty `Platform{}`

**Review by abyrne55 — CHANGES_REQUESTED** [review]
Good stuff, just requested an additional test case

File: pkg/data/cloud/platform_test.go (hunk 1 above)
Comment (joshbranham) [reply pkg/data/cloud/platform_test.go:125]: Good call, that failed so added a fix for that scenario as well.

**Review by abyrne55 — APPROVED** [review]
Tested locally with `./osd-network-verifier egress --platform=aws-hosted-cp` and confirmed new alias selects the aws-hcp egress list as expected. Also tested `--platform=""` and confirmed error.

Nice work, @joshbranham!

/lgtm
